clients/src/test/java/org/apache/kafka/clients/producer/MockProducerTest.java

```diff
@@ -77,6 +76,7 @@ public void testPartitioner() throws Exception {
         assertEquals("Partition should be correct", 1, metadata.get().partition());
         producer.clear();
         assertEquals("Clear should erase our history", 0, producer.history().size());
+        producer.close();
```

Comment:
Don't think readability would suffer. For `MockProducer` is might also not be bad atm if `close()` is not called in case the test fails with an exception and thus `close()` is never called. It just feel "right" to make sure to call `close()` for all classes that are `Closable` -- even in case of failure. This might also be more "future prove" in case `close()` implementation changes at some point. But I also don't insist on the change.